src/object_store/src/object/opendal_engine/fs.rs

```diff
@@ -25,6 +25,7 @@ impl OpendalObjectStore {
         let mut builder = Fs::default();
 
         builder.root(&root);
+        builder.atomic_write_dir(&root);
```

Comment:
I don't think passing the same directory is correct. 😅 Please test the PR before rushing it out.

Reference: https://github.com/apache/opendal/pull/991

Reply:
> I don't think passing the same directory is correct. 😅 Please test the PR before rushing it out.
> 
> Reference: [apache/opendal#991](https://github.com/apache/opendal/pull/991)

+1

Reply:
I think we can build a independent path for applying this setting when creating builder, just like [cluster_id](https://github.com/risingwavelabs/risingwave/blob/4197ad50c766c8d086d29948ef34e13582e72251/src/meta/src/hummock/manager/mod.rs#L3351),  and make sure this directory is not used.